doc/api-migration.md

```diff
@@ -1,4 +1,85 @@
-## Migrating to the new API
+# Migrating to new APIs
+
+## 0.17.1 &rarr; 0.18.0
```

Comment:
I don't know what the rules about markdown and HTML entities are, but can we just enter these characters directly, for at least two reasons:
- These days you can pretty much assume UTF-8 everywhere
- The character will render properly in plaintext viewers